runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -211,6 +211,372 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
    return node->getRegister();
    }
 
+/**
+ * \brief
+ *
+ * Use vector instructions to find the index of a sub-string inside a string assuming both strings have the same element size.
+ * Each element is 1-byte for compact strings and 2-bytes for non-compressed strings.
+ *
+ * \details
+ *
+ * The vector sequence searches for the first character of the sub-string inside the source/master string. If the first character
+ * is located, it'll perform iterative vector binary compares to match the rest of the sub-string starting from the first character position.
+ *
+ *
+ * This evaluator inlines the following Java intrinsic methods:
+ *
+ * <verbatim>
+ * For Java 9 and above:
+ *
+ * StringLatin1.indexOf(s1Value, s1Length, s2Value, s2Length, fromIndex);
+ * StringUTF16.indexOf(s1Value, s1Length, s2Value, s2Length, fromIndex);
+ *
+ * For Java 8:
+ * com.ibm.jit.JITHelpers.intrinsicIndexOfStringLatin1(char[] s1Value, int s1len, char[] s2Value, int s2len, int start);
+ * com.ibm.jit.JITHelpers.intrinsicIndexOfStringUTF16(char[] s1Value, int s1len, char[] s2Value, int s2len, int start);
+ *
+ * Assumptions:
+ *
+ * -# 0 <= fromIndex < s1Length
+ * -# s1Length could be anything: positive, negative or 0.
+ * -# s2Length > 0
+ * -# s1Value and s2Value are non-null arrays and are interpreted as byte arrays.
+ * -# s1Length and s2Length are not related. i.e. s1Length could be smallers than s2Length.
+ *
+ * <\verbatim>
+ *
+ *
+ * \param node the intrinsic function call node
+ * \param cg the code generator
+ * \param elementSize element size in bytes. This is 1 for compact strings of Latin1 letters, and it's 2 for non-compact strings.
```

Comment:
I would avoid giving user the option here. The evaluator only seems to support `1` and `2` so why even give the user the option to accidentally pass `4` for example? Why not follow the footsteps of `intrinsicIndexOf` and supply a boolean dictating whether the `char` is compressed or not? This would also add symmetry to the APIs.

Reply:
Changed the `elementSize ` parameter to a boolean `isCompact`.